itest/lnd_estimate_route_fee_test.go

```diff
@@ -0,0 +1,75 @@
+package itest
+
+import (
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/stretchr/testify/require"
+)
+
+// testEstimateRouteFee tests that we are able to successfully route a
+// payment using multiple shards across different paths.
+func testEstimateRouteFee(ht *lntest.HarnessTest) {
+	mts := newMppTestScenario(ht)
+
+	// Set up a network with three different paths Alice <-> Bob. Channel
+	// capacities are set such that the payment can only succeed if (at
+	// least) three paths are used.
+	//
+	//              _ Eve _
+	//             /       \
+	// Alice -- Carol ---- Bob
+	//      \              /
+	//       \__ Dave ____/
+	//
+	req := &mppOpenChannelRequest{
+		amtAliceCarol: 235_000,
+		amtAliceDave:  135_000,
+		amtCarolBob:   135_000,
+		amtCarolEve:   135_000,
+		amtDaveBob:    135_000,
+		amtEveBob:     135_000,
+	}
+	mts.openChannels(req)
+	chanPointAliceDave := mts.channelPoints[1]
+
+	// Increase Dave's fee to make the test deterministic. Otherwise, it
+	// would be unpredictable whether pathfinding would go through Charlie
+	// or Dave for the first shard.
+	expectedPolicy := mts.updateDaveGlobalPolicy()
+
+	// Make sure Alice has heard it.
+	ht.AssertChannelPolicyUpdate(
+		mts.alice, mts.dave, expectedPolicy, chanPointAliceDave, false,
```

Comment:
I think this is still correct as we try to ensure that Alice sees the high fee adjustment of Dave, so that we can be confident that our probe runs through Alice-Carol-Bob.